CMakeLists.txt

```diff
@@ -297,22 +297,22 @@ if(CPPDAP_BUILD_TESTS)
 
     if(CPPDAP_USE_EXTERNAL_GTEST_PACKAGE)
         find_package(GTest REQUIRED)
+        set(CPPDAP_GTEST_TARGET GTest::gtest)
     else()
-        list(APPEND DAP_TEST_LIST
-            ${CPPDAP_GOOGLETEST_DIR}/googletest/src/gtest-all.cc
-        )
-
-        set(DAP_TEST_INCLUDE_DIR
-            ${CPPDAP_GOOGLETEST_DIR}/googlemock/include/
-            ${CPPDAP_GOOGLETEST_DIR}/googletest/
-            ${CPPDAP_GOOGLETEST_DIR}/googletest/include/
-        )
+        # Build gtest from source
+        set(INSTALL_GTEST OFF)
+        add_subdirectory(${CPPDAP_GOOGLETEST_DIR}/googlemock)
+        # googletest has -Werror=maybe-uninitialized problem.
+        # Disable all warnings in googletest code.
+        target_compile_options(gtest PRIVATE -w)
+        set(CPPDAP_GTEST_TARGET gtest)
+        set(DAP_TEST_INCLUDE_DIR ${CPPDAP_GOOGLETEST_DIR}/googlemock/include/)
```

Comment:
Turns out that this isn't necessary. You need to also link against the `gmock` target. See, `GTest::gtest` is an alias for those two targets, but sadly, it doesn't get added with `add_subdirectory`.

Also, you need to enable `gtest_force_shared_crt` on Windows, anyway, otherwise you get linker errors about mismatched CRTs. Finally, when setting sub-project variables in CMake, you want to set them in the CACHE, because these only get set once during CMake generation, but any build steps that rely on these values will not work properly if they're not cached (CMake sucks).

Take a look at my changes here: https://github.com/google/cppdap/compare/main...amaiorano:cppdap:main#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR299

Reply:
I pulled in your commit, thanks!

I still need to add the -w compile option though. Am trying that.